modules/nextflow/src/main/groovy/nextflow/k8s/client/K8sClient.groovy

```diff
@@ -338,6 +338,25 @@ class K8sClient {
      *      the second element is the text (json) response
      */
     protected K8sResponseApi makeRequest(String method, String path, String body=null) throws K8sResponseException {
+
+        int remainingTrials = 5
+
+        while ( remainingTrials > 0 ) {
+            remainingTrials--
+
+            try {
+                return makeRequestCall( method, path, body )
+            } catch ( SocketException e ) {
+                log.error "[K8s] API request throw socket exception: $e.localizedMessage for $method $path ${body ? '\n'+prettyPrint(body).indent() : ''}"
+                if ( remainingTrials ) log.info( "[K8s] Try API request again, remaining trials: $remainingTrials" )
+                else throw e
+                sleep( 1500 )
```

Comment:
An exponential backoff retry could be a better choice, here there's an example 

http://github.com/nextflow-io/nextflow/blob/f0ccfc7e83269bae6734fcafabf231a7c90e2087/modules/nextflow/src/main/groovy/nextflow/util/SimpleHttpClient.groovy#L173-L175